### PR TITLE
fix compilation without rocblas

### PR DESCRIPTION
**Pull request by apwojcik**
Fix the compilation while `MIOPEN_USE_ROCBLAS` is `OFF`. We need to strip off the rocblas library in UAI, so this one is important.

### CODE REVIEW

File: src/solver/mha/mha_solver_backward.cpp [hunk 1]
```diff
@@ -90,8 +90,8 @@ bool MhaBackward::IsApplicable([[maybe_unused]] const ExecutionContext& context,
 
     auto [N, H, S, D] = miopen::tien<4>(descsBwd.kDesc.GetLengths());
 
-    return MIOPEN_USE_GEMM                                            //
-           && !miopen::IsDisabled(ENV(MIOPEN_DEBUG_ATTN_NAIVE_BWD))   //
+#if MIOPEN_USE_ROCBLAS
```
Comment (CAHEK7) [inline src/solver/mha/mha_solver_backward.cpp]: It does not contain any rocblas-specific functionality, except the `USE_ROCBLAS_EX3` which should be properly defined with or without rocblas.
And currently, prior to #3030 `MIOPEN_USE_GEMM` is equal to `MIOPEN_USE_ROCBLAS` so it is not applicable and must not be excluded in that way.

**Review by CAHEK7 — CHANGES_REQUESTED** [review]
Could you please explain what is the problem with `isApplicable` methods?

File: src/solver/mha/mha_solver_backward.cpp (hunk 1 above)
Comment (apwojcik) [reply src/solver/mha/mha_solver_backward.cpp]: When `MIOPEN_USE_GEM` (aka `MIOPEN_USE_ROCBLAS`) is 0, the compiler warns that the rest of the expression has no effect. When `-Werror` is additionally being set, compilation breaks.

Comment (apwojcik) [reply src/solver/mha/mha_solver_backward.cpp]: Here is the effect of `MIOPEN_USE_GEMM` equal to `0` (false) when compiling.

```
C:\...\MIOpen\src\solver\mha\mha_solver_backward.cpp:115:17: warning: code will never be executed [-Wunreachable-code]
  115 |            && ((descsBwd.kDesc.GetType() == miopenFloat)              //
      |                 ^~~~~~~~
C:\...\MIOpen\src\solver\mha\mha_solver_backward.cpp:114:15: warning: code will never be executed [-Wunreachable-code]
  114 |            && descsBwd.kDesc.GetType() == descsBwd.dvDesc.GetType()   //
      |               ^~~~~~~~
C:\...\MIOpen\src\solver\mha\mha_solver_backward.cpp:113:15: warning: code will never be executed [-Wunreachable-code]
  113 |            && descsBwd.kDesc.GetType() == descsBwd.dkDesc.GetType()   //
      |               ^~~~~~~~
C:\...\MIOpen\src\solver\mha\mha_solver_backward.cpp:112:15: warning: code will never be executed [-Wunreachable-code]
  112 |            && descsBwd.kDesc.GetType() == descsBwd.dqDesc.GetType()   //
...
```
There are about 40 of these kinds of messages in a row.

Comment (CAHEK7) [reply src/solver/mha/mha_solver_backward.cpp]: Ahh, but it is intentional. It looks like false-positive for me.
Ok, don't want fighting with the compiler, let's just exclude it by ifdef. But there will be a merge conflict with #3030

**Review by CAHEK7 — APPROVED** [review]

File: src/ocl/rnnocl.cpp [hunk 2]
```diff
@@ -1078,8 +1082,6 @@ void RNNDescriptor::RNNForwardMS(Handle& handle,
     (void)y;
     (void)hy;
     (void)cy;
-    (void)reserveSpace;
```
Comment (junliume) [inline src/ocl/rnnocl.cpp:1081]: @apwojcik why these two lines of changes?

Comment (junliume) [reply src/ocl/rnnocl.cpp:1081]: We can let it go as long as hiptidy is not complaining, but it looks like not necessary.

**Review by junliume — APPROVED** [review]

**Pull request merged**